.jenkins/update_timestamps.py

```diff
@@ -0,0 +1,57 @@
+import re
+import sys
+import os
+import subprocess
+
+def get_last_commit_timestamp_for_file(file_path: str) -> str:
+    """Get the last commit timestamp for a file.
+
+    Args:
+        file_path (str): Path to file
+
+    Returns:
+        str: Last committed timestamp string
+    """
+    git_command = ["git", "log", "-1", "--format=%at", "--", file_path]
+    timestamp = subprocess.check_output(git_command).decode().strip()
+
+    if not timestamp:
+        # If there is no git commit history, use last modified date
+        timestamp = str(int(os.path.getmtime(file_path)))
+
+    date_command = ["date", "-d", "@" + timestamp, "+%I:%M %p, %B %d, %Y"]
+    return subprocess.check_output(date_command).decode().strip()
+
+def update_timestamp(file_path: str):
+    """Adds a timestamp of the most recent time the file was edited.
+
+    Args:
+        file_path (str): Path to file
+    """
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    # Get current timestamp
+    timestamp = get_last_commit_timestamp_for_file(file_path)
+    timestamp_line = f'{"# " if file_path.endswith("py") else ""}**Updated:** *{timestamp}*\n'
+    timestamp_pattern = r'\*\*Updated:\*\*\s\**\d{1,2}:\d{2} [AP]M, \w+ \d{1,2}, \d{4}\*'
```

Comment:
I would rather try to avoid regexes like this

Reply:
I updated it! For my own knowledge, may I ask why? Is it because it's too specific? I was trying to be specific just so I didn't grab any lines unexpectedly from the tutorial files.

Reply:
I also think it looks better if it's right after the title, but this comes with some caveats. Since we do not standardize the title across .py and .rst files, I would have to get very broad. Many people start files in many different ways. Often they follow the line with a separator like === or *** but some files don't have this. I would love your input, the best I came up with was "Define title as first line with alphabet characters, except in .rst files that begin with `". Is that an acceptable definition of "title"?